server/pkg/api/cluster_apps.go

```diff
@@ -131,6 +126,14 @@ func (s *Server) configureSSOForClusterApps(ctx context.Context, orgId, clusterI
 		return fmt.Errorf("failed to get cluster app launches from cluster %s, err: %v", clusterID, resp)
 	}
 
+	if err := s.serverStore.InsertCacheAppLaunches(orgId, clusterID, resp.LaunchConfigList); err != nil {
+		return fmt.Errorf("failed to store cluster app launches on server db %s, err: %v", clusterID, err)
+	}
+
+	s.mutex.Lock()
+	s.orgClusterIDCache[orgId+"-"+clusterID] = time.Now().Add(delayTimeinMin * time.Minute).Unix()
+	s.mutex.Unlock()
```

Comment:
cache is not needed, it should always fetch from server DB
update same in all places